### PR TITLE
feat: new aggregate snapshotting + migration event and command

### PR DESCRIPTION
**Pull request by jvandaal**

### CODE REVIEW

File: src/ParcelRegistry/Legacy/CommandHandlerModules.cs [hunk 1]
```diff
@@ -11,6 +11,9 @@ public static void Register(ContainerBuilder containerBuilder)
                 .RegisterType<ParcelProvenanceFactory>()
                 .SingleInstance();
 
+            containerBuilder.RegisterType<FixGrar1475ProvenanceFactory>().AsSelf();
+            containerBuilder.RegisterType<FixGrar1637ProvenanceFactory>().AsSelf();
```
Comment (jvandaal) [inline src/ParcelRegistry/Legacy/CommandHandlerModules.cs:15]: Added them here based upon https://github.com/Informatievlaanderen/building-registry/blob/main/src/BuildingRegistry/Legacy/CommandHandlerModules.cs#L18-L20

Don't know how they were previously resolved (and if there's a conflict now)?

File: src/ParcelRegistry/Parcel/ValueObjects/VbrCaPaKey.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+{
+    using System;
+    using Be.Vlaanderen.Basisregisters.AggregateSource;
+    using Be.Vlaanderen.Basisregisters.Generators.Guid;
+    using Be.Vlaanderen.Basisregisters.GrAr.Common;
+    using Newtonsoft.Json;
+
+    public class VbrCaPaKey : StringValueObject<VbrCaPaKey>
+    {
+        private static readonly Guid Namespace = new Guid("28068113-774a-4f6f-8264-9ddb39095bd4");
```
Comment (jvandaal) [inline src/ParcelRegistry/Parcel/ValueObjects/VbrCaPaKey.cs:11]: @ArneD we have copy-pasted this class. So the Guid is the same as in Legacy. Should it be the same as in legacy?

File: src/ParcelRegistry/Parcel/Commands/MigrateParcel.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        public bool IsRemoved { get; }
+        public List<AddressPersistentLocalId> AddressPersistentLocalIds { get; }
+
+        public Provenance Provenance { get; }
+
+        public MigrateParcel(
+            Legacy.ParcelId parcelId,
+            Legacy.ParcelStatus parcelStatus,
+            bool isRemoved,
+            IEnumerable<AddressPersistentLocalId> addressPersistentLocalIds,
```
[the remark this answers is not in the document]
Comment (ArneD) [reply src/ParcelRegistry/Parcel/Commands/MigrateParcel.cs:25]: Yes!

File: src/ParcelRegistry/Parcel/SnapshotContainerV2.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+namespace ParcelRegistry.Parcel
+{
+    using Be.Vlaanderen.Basisregisters.AggregateSource.Snapshotting;
+
+    public class SnapshotContainerV2 : SnapshotContainer
+    {
```
Comment (ArneD) [inline src/ParcelRegistry/Parcel/SnapshotContainerV2.cs:6]: remove empty line => `{ }`

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/ParcelRegistry/Parcel/ValueObjects/VbrCaPaKey.cs (hunk 2 above)
Comment (jvandaal) [reply src/ParcelRegistry/Parcel/ValueObjects/VbrCaPaKey.cs]: Done.

File: src/ParcelRegistry/Parcel/SnapshotContainerV2.cs (hunk 4 above)
Comment (jvandaal) [reply src/ParcelRegistry/Parcel/SnapshotContainerV2.cs]: Done

**Review by ArneD — APPROVED** [review]

File: src/ParcelRegistry/Legacy/CommandHandlerModules.cs (hunk 1 above)
Comment (jvandaal) [reply src/ParcelRegistry/Legacy/CommandHandlerModules.cs:15]: Just saw that ParcelCommandHandlerModule is also created manually in import. Didn't notice that before.

**Pull request merged**